Classes/Service/NodeRedirectService.php

```diff
@@ -303,14 +306,23 @@ protected function createUriPathsAcrossDimensionsForNode(string $nodeIdentifier,
      */
     protected function hasNodeUriChanged(NodeInterface $node, Workspace $targetWorkspace): bool
     {
-        $newUriPath = $this->buildUriPathForNode($node);
-        $newUriPath = $this->removeContextInformationFromRelativeNodeUri($newUriPath);
-
         $nodeInTargetWorkspace = $this->getNodeInWorkspace($node, $targetWorkspace);
         if (!$nodeInTargetWorkspace) {
             return false;
         }
-        $oldUriPath = $this->buildUriPathForNode($nodeInTargetWorkspace);
+        try {
+            $newUriPath = $this->buildUriPathForNode($node);
+        } catch (\Exception $exception) {
+            $this->logger->info(sprintf('Failed to build new URI for updated node "%s": %s', $node->getContextPath(), $exception->getMessage()));
+            return false;
+        }
+        $newUriPath = $this->removeContextInformationFromRelativeNodeUri($newUriPath);
+        try {
+            $oldUriPath = $this->buildUriPathForNode($nodeInTargetWorkspace);
+        } catch (\Exception $exception) {
+            $this->logger->info(sprintf('Failed to build previous URI for updated node "%s": %s', $node->getContextPath(), $exception->getMessage()));
```

Comment:
I'm not sure about the log level

Reply:
Info would probably not appear in production? Isn't a warning better?

Reply:
that case is logged in the `FrontendNodeRoutePartHandler` too. Besides, that's not actually a very exceptional case for shortcut nodes (see https://github.com/neos/neos-development-collection/issues/3256). I think "warning" would be too high